templates/addons/windows/calico/calico.yaml

```diff
@@ -0,0 +1,238 @@
+            - name: CNI_NETWORK_CONFIG
+              valueFrom:
+                configMapKeyRef:
+                  name: calico-config-windows
+                  key: cni_network_config
+            # Set the hostname based on the k8s node name.
+            - name: KUBERNETES_NODE_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: spec.nodeName
+            # CNI MTU Config variable
+            - name: CNI_MTU
+              valueFrom:
+                configMapKeyRef:
+                  name: calico-config-windows
+                  key: veth_mtu
+            # Prevents the container from sleeping forever.
+            - name: SLEEP
+              value: "false"
+            - name: K8S_SERVICE_CIDR
+              value: "10.96.0.0/12"
+          volumeMounts:
+            - mountPath: /host/opt/cni/bin
+              name: cni-bin-dir
+            - mountPath: /host/etc/cni/net.d
+              name: cni-net-dir
+            - name: kubeadm-config
+              mountPath: /etc/kubeadm-config/
+          securityContext:
+            windowsOptions:
+              hostProcess: true
+              runAsUserName: "NT AUTHORITY\\system"
+      containers:
+      - name: calico-node-startup
+        image: sigwindowstools/calico-node:v3.20.0-hostprocess
+        args: ["$env:CONTAINER_SANDBOX_MOUNT_POINT/calico/node-service.ps1"]
+        workingDir: "$env:CONTAINER_SANDBOX_MOUNT_POINT/calico/"
+        imagePullPolicy: Always
+        volumeMounts:
+        - name: calico-config-windows
+          mountPath: /etc/kube-calico-windows/
+        env:
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.name
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.namespace
+        - name: CNI_IPAM_TYPE
+          value: "calico-ipam"
+        - name: CALICO_NETWORKING_BACKEND
+          value: "vxlan"
+        - name: KUBECONFIG
+          value: "C:/etc/cni/net.d/calico-kubeconfig"
+        - name: VXLAN_VNI
+          value: "4096"
+      - name: calico-node-felix
+        image: sigwindowstools/calico-node:v3.20.0-hostprocess
+        args: ["$env:CONTAINER_SANDBOX_MOUNT_POINT/calico/felix-service.ps1"]
+        imagePullPolicy: Always
+        workingDir: "$env:CONTAINER_SANDBOX_MOUNT_POINT/calico/"
+        volumeMounts:
+        - name: calico-config-windows
+          mountPath: /etc/kube-calico-windows/
+        env:
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.name
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.namespace
+        - name: VXLAN_VNI
+          value: "4096"
+        - name: KUBECONFIG
+          value: "C:/etc/cni/net.d/calico-kubeconfig"
```

Comment:
could this reference a config map to save having to save the secret on the c: outside of the container root path?  I've been doing something similar, but it does need copying out to root for kube-proxy as that seems to be hardcoded to look in c:/var/lib/kube-proxy

Reply:
This config file is for the calico.exe cni binary, which requires access to the api server and is invoked by CRI (via containerd) so I believe there is no way to avoid having this file.   It is re-used be the Felix service but we could potentially use the service account felix is linked too (but it has the same RBAC as the kubeconfig is generated by using the same SA)

Reply:
yeah, I think I have found this too especially with kube-proxy, I guess I was just hoping that the container could write this file itself from a secret rather than have to provide it manually? unless it already is and I'm missing something?

Reply:
id say merge and iterate a pluggable file path